From ad23c84ca38e828004707a1ab597d7674dbc65f0 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: Brendan Higgins <brendanhiggins@google.com>
Cc: David Gow <davidgow@google.com>
Cc: damon@lists.linux.dev
Cc: linux-mm@kvack.org
Cc: kunit-dev@googlegroups.com
Cc: linux-kselftest@vger.kernel.org
Cc: linux-kernel@vger.kernel.org
Date: Wed, 17 Jul 2024 16:36:01 -0700
Subject: [PATCH 8/9] mm/damon: move kunit tests to tests/ subdirectory with
 _kunit suffix

There was a discussion about better places for kunit test code[1] and
test file name suffix[2].  Folowwing the conclusion, move kunit tests
for DAMON to mm/damon/tests/ subdirectory and rename those.

[1] https://lore.kernel.org/CABVgOS=pUdWb6NDHszuwb1HYws4a1-b1UmN=i8U_ED7HbDT0mg@mail.gmail.com
[2] https://lore.kernel.org/CABVgOSmKwPq7JEpHfS6sbOwsR0B-DBDk_JP-ZD9s9ZizvpUjbQ@mail.gmail.com

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c                                | 2 +-
 mm/damon/dbgfs.c                               | 2 +-
 mm/damon/sysfs.c                               | 2 +-
 mm/damon/{core-test.h => tests/core-kunit.h}   | 0
 mm/damon/{dbgfs-test.h => tests/dbgfs-kunit.h} | 0
 mm/damon/{sysfs-test.h => tests/sysfs-kunit.h} | 0
 mm/damon/{vaddr-test.h => tests/vaddr-kunit.h} | 0
 mm/damon/vaddr.c                               | 2 +-
 8 files changed, 4 insertions(+), 4 deletions(-)
 rename mm/damon/{core-test.h => tests/core-kunit.h} (100%)
 rename mm/damon/{dbgfs-test.h => tests/dbgfs-kunit.h} (100%)
 rename mm/damon/{sysfs-test.h => tests/sysfs-kunit.h} (100%)
 rename mm/damon/{vaddr-test.h => tests/vaddr-kunit.h} (100%)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index a1c32becfc73..1d2c4daef157 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2275,4 +2275,4 @@ static int __init damon_init(void)
 
 subsys_initcall(damon_init);
 
-#include "core-test.h"
+#include "tests/core-kunit.h"
diff --git a/mm/damon/dbgfs.c b/mm/damon/dbgfs.c
index 51a6f1cac385..b4213bc47e44 100644
--- a/mm/damon/dbgfs.c
+++ b/mm/damon/dbgfs.c
@@ -1145,4 +1145,4 @@ static int __init damon_dbgfs_init(void)
 
 module_init(damon_dbgfs_init);
 
-#include "dbgfs-test.h"
+#include "tests/dbgfs-kunit.h"
diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index ecf0835e34b8..4daac92be30b 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1908,4 +1908,4 @@ static int __init damon_sysfs_init(void)
 }
 subsys_initcall(damon_sysfs_init);
 
-#include "sysfs-test.h"
+#include "tests/sysfs-kunit.h"
diff --git a/mm/damon/core-test.h b/mm/damon/tests/core-kunit.h
similarity index 100%
rename from mm/damon/core-test.h
rename to mm/damon/tests/core-kunit.h
diff --git a/mm/damon/dbgfs-test.h b/mm/damon/tests/dbgfs-kunit.h
similarity index 100%
rename from mm/damon/dbgfs-test.h
rename to mm/damon/tests/dbgfs-kunit.h
diff --git a/mm/damon/sysfs-test.h b/mm/damon/tests/sysfs-kunit.h
similarity index 100%
rename from mm/damon/sysfs-test.h
rename to mm/damon/tests/sysfs-kunit.h
diff --git a/mm/damon/vaddr-test.h b/mm/damon/tests/vaddr-kunit.h
similarity index 100%
rename from mm/damon/vaddr-test.h
rename to mm/damon/tests/vaddr-kunit.h
diff --git a/mm/damon/vaddr.c b/mm/damon/vaddr.c
index 58829baf8b5d..b0e8b361891d 100644
--- a/mm/damon/vaddr.c
+++ b/mm/damon/vaddr.c
@@ -730,4 +730,4 @@ static int __init damon_va_initcall(void)
 
 subsys_initcall(damon_va_initcall);
 
-#include "vaddr-test.h"
+#include "tests/vaddr-kunit.h"
-- 
2.39.2

